### PR TITLE
Renable running Arm64 test cases in CI

### LINKED ISSUE #83947: Arm64 tests are not run as part of CI
In https://github.com/dotnet/runtime/pull/76642 , we added checks to exclude Arm related projects based on `TargetArchitecture`.

https://github.com/dotnet/runtime/blob/922c75c837fc95092e1d9e433a087ac62082953d/src/tests/JIT/HardwareIntrinsics/HardwareIntrinsics_ro.csproj#L7-L8

However, when we build test managed components for AnyOS, we end up passing `archType=x64` which translates to `/p:TargetArchitecture=x64`:

https://github.com/dotnet/runtime/blob/922c75c837fc95092e1d9e433a087ac62082953d/eng/pipelines/common/templates/runtimes/build-test-job.yml#L122

This can be seen from the logs of `coreclr Common Pri0 Test Build AnyOS AnyCPU checked` job:

```
2023-03-24T16:47:53.7325522Z /__w/1/s/src/tests/build.sh -log:Managed allTargets skipnative skipgeneratelayout skiptestwrappers checked x64     ci /p:TargetOS=AnyOS
2023-03-24T16:47:53.7326524Z ========================== Starting Command Output ===========================
2023-03-24T16:47:53.7345342Z [command]/usr/bin/bash --noprofile --norc /__w/_temp/249d4290-e028-48ef-84f2-448c189a994c.sh
2023-03-24T16:47:53.7629393Z __DistroRid: linux-x64
2023-03-24T16:47:53.7657493Z Building Tests...
2023-03-24T16:47:53.7686206Z __TargetOS: linux
2023-03-24T16:47:53.7686828Z __TargetArch: x64
2023-03-24T16:47:53.7687367Z __BuildType: Checked
2023-03-24T16:47:53.7688000Z __TestIntermediatesDir: /__w/1/s/artifacts/tests/coreclr/obj/linux.x64.Checked
2023-03-24T16:47:53.7688926Z __NativeTestIntermediatesDir: /__w/1/s/artifacts/tests/coreclr/obj/linux.x64.Checked/Native
2023-03-24T16:47:53.7689874Z __ManagedTestIntermediatesDir: /__w/1/s/artifacts/tests/coreclr/obj/linux.x64.Checked/Managed
2023-03-24T16:47:53.7690720Z Creating TestBinDir: /__w/1/s/artifacts/tests/coreclr/linux.x64.Checked
2023-03-24T16:47:53.7696258Z Creating LogsDir: /__w/1/s/artifacts/log

```

As a result, we do not include `Arm` test cases in CI run. I downloaded the `HardwareIntrinsics_ro.dll` and checked using ILSpy and I don't see `AdvSimd.dll` project being referenced.

image

### PR DESCRIPTION
**Pull request by kunalspathak**
The original change prohibited running Arm tests from non-Arm platform, however, we build AnyOS/AnyConfiguration tests and we don't know what the targeted CI leg is for. For now, just remove such restriction and see if it picks up Arm test cases. After that, probably, add a property (or look for a property, if there is one available) that can be used to exclude the Arm test cases from getting included in non-arm platforms.

Fixes https://github.com/dotnet/runtime/issues/83947

### CODE REVIEW

**Comment by kunalspathak** [comment]
These are included now:

image

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Comment by tannergooding** [comment]
This likely needs to include the logic to continue filtering these on irrelevant platforms. The filtering was done due to the timeouts that otherwise get hit, particularly in stress jobs

**Review by tannergooding — APPROVED** [review]

**Review by trylek — APPROVED** [review]
LGTM, thanks Kunal! Just to clarify test priority checks that the build script refers to at line 8 and 10 should still be legal, we aren't trying to build Pri0 and Pri1 tests at the same time (there's actually no way to do it now considering they're used in different pipelines); other than that you're absolutely right that the architecture-specific checks shouldn't be put in the primary <code>csproj</code> scripts exactly as you described.

**Comment by kunalspathak** [comment]
> you're absolutely right that the architecture-specific checks shouldn't be put in the primary `csproj` scripts exactly as you described.

What is a good way to accomplish this if we want to not skip building certain `csproj` files based on architecture?

**Comment by trylek** [comment]
As I described in a comment on the issue thread

https://github.com/dotnet/runtime/issues/83980

the current POR is basically to use the <code>CLRTestTargetUnsupported</code> property for tests applicable to only a certain OS / architecture or build / execution flavor (e.g. some tests are incompatible with GC stress, IJW tests are currently incompatible with Crossgen2 as it doesn't support mixed managed & native code etc.). I can easily imagine there may be previously overlooked inconsistencies in application of this rule. Please let me know if you believe there are cases that cannot leverage this technique so that we'd need to invent a new mechanism. For instance, the new-style merged test wrappers don't yet support any equivalent of this logic so that as a temporary measure we mark all architecture / OS-specific tests as "requiring process isolation" even though no isolation is really needed, all that's needed is for the merged wrapper to stop trying to directly call the tests in question and instead resort to the legacy way of running the test script. While not ideal, it thankfully applies to a tiny fraction of the entire test set. Once the bulk of test merging is behind us, we can revisit these and try to implement more performant solutions. That may require additional JIT work as one of the problems we hit previously was that JITting the merged test wrapper calling a test "for the wrong architecture" crashed JIT on an assertion failure despite the fact that we would have ultimately skipped execution of the test.

**Comment by BruceForstall** [comment]
It seems like with this change we'll run a **lot** more tests in the CI (some, of course, we should be running and currently aren't).

Should `NumberOfStripesToUseInStress` be increased to compensate?

Also, with this change, the comments:
```
    <!-- We have a lot of tests here so run them in outerloop on platforms where they aren't accelerated -->
    <!-- For most of these, that just involves excluding the projects when the architecture is mismatched -->
```
are obsolete and can be removed.

Also,
```
    <!-- For Vector512, we only have a very small pool of machines with acceleration support, so they are always outerloop -->
```
is also obsolete (all our Helix xarch machines support AVX-512).

**Comment by tannergooding** [comment]
> Just to clarify test priority checks that the build script refers to at line 8 and 10 should still be legal, we aren't trying to build Pri0 and Pri1 tests at the same time (there's actually no way to do it now considering they're used in different pipelines); other than that you're absolutely right that the architecture-specific checks shouldn't be put in the primary csproj scripts exactly as you described.

@trylek, the consideration here is that we have a lot of hardware intrinsic tests. All tests "can" run on any machine. However, the test pattern on mismatched hardware is validation that a `PlatformNotSupportedException` is thrown.

For `Pri1` we want to run "everything", regardless of target architecture. For `Pri0` we want to only run the tests that are "possible supported" by the underlying hardware.

So, for `Pri0` on `Arm64` we want to run the https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/Arm tests. But we don't want to run the https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/X86 tests. We want to run both for `Pri1`.

Inversely for `Pri0` on x86 Windows and x64 Windows/Linux/MacOS we want to run https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/X86 but don't want to run  https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/Arm. We again want to run both sets for `Pri1`.

Given this, is there a setup that would allow this to be achieved?

**Comment by kunalspathak** [comment]
We will explore the ideas in a separate PR. I will merge this one to unblock the Arm64 testing.

**Pull request merged**
